ouroboros-network-mock/src/Ouroboros/Network/Mock/Chain.hs

```diff
@@ -225,14 +225,17 @@ rollback p (c :> b) | blockPoint b == p = Just (c :> b)
 rollback p Genesis  | p == genesisPoint = Just Genesis
                     | otherwise         = Nothing
 
+-- | Find the provided point's successor on the provided chain. If the point is
+-- not on the chain, this function's behavior is undefined. Use 'pointOnChain'
+-- to check whether a point is on a chain.
```

Comment:
This is, more or less, the function's de facto contract already. Use sites of this function take care to guard against the `error` by checking membership before calling it, so I've codified calling it without doing so as undefined behavior and then redefined the behavior to eliminate the `error`. This warrants attention, though, since it does technically represent a change in semantics.

Reply:
I can do that. However, another option would be to change this function's type to reflect this contract. Instead of
```hs
successorBlock :: HasHeader block => Point block -> Chain block -> Maybe block
```
we would have
```hs
successorBlock :: HasHeader block => Point block -> Chain block -> Maybe (Maybe block)
```
or some other three-pronged type. I gamed this out and don't think it would introduce any new `error`-like behavior at existing call sites to catch the new case - though one site already guards against this case with an `assert`. What do you think?

Reply:
`Maybe (Maybe ...)` is something confusing, it's not obvious which `Nothing` case is which one, and thus a bit error prone.  I'd leave it as is.

Reply:
Ok, fair enough, I'll leave it. Regarding making an issue for updating the call sites, I don't think that's necessary - there appear to be precisely three, and all of them guard against calling `successorBlock` in cases where the point isn't on the chain.

Would you prefer that the `error` behavior remain, or that we keep my change to overloading `Nothing`?

Reply:
Let's leave it as it was.

Reply:
Can do.

Reply:
I restored the function's original error behavior, but I think its error condition still ought to be documented. I want to make sure I'm using chain/block/tip vocabulary appropriately - what do you think of:
```hs
-- | A 'Just' result holds the provided point's successor block. A 'Nothing'
-- result means the provided point was the tip.
--
-- The function will error if the point is not on the chain - callers can use
-- 'pointOnChain' to check point membership on a chain in advance of calling
-- this function.
```

Reply:
LGTM